src/frequenz/sdk/_data_ingestion/battery_utils.py

```diff
@@ -0,0 +1,361 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+"""Class to store features for battery."""
+import asyncio
+import logging
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from enum import Enum
+from typing import Generic, Optional, Set, TypeVar, Union
+
+import pytz
+from frequenz.channels import Peekable
+
+from ..microgrid import ComponentGraph
+from ..microgrid import get as get_microgrid
+from ..microgrid.component import (
+    BatteryData,
+    ComponentCategory,
+    ComponentData,
+    InverterData,
+)
+from ..microgrid.component._component_errors import ErrorLevel
+from ..microgrid.component._component_states import (
+    BatteryRelayState,
+    BatteryState,
+    InverterState,
+)
+
+# Time needed to get first component data.
+# Constant for now. In future should be configurable in UI.
+_START_DELAY_SEC = 2.0
+
+_logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+
+
+class WorkingState(Enum):
+    """Tells if battery is can be used."""
+
+    # Component it is not working.
+    NOT_WORKING = 0
+    # Component should work, but it failed in last request. It is blocked for few
+    # seconds and it is not recommended to use it until it is necessary.
+    UNCERTAIN = 1
+    # Component is working
+    WORKING = 2
+
+
+@dataclass
+class _ComponentReceiver(Generic[T]):
+    component_id: int
+    receiver: Peekable[T]
+
+
+class BatteryUtils:
```

Comment:
For the other class also, it is also not really the data-pipeline BatteryPool that we have been talking about,  but something specific to the the PowerDistributor: https://github.com/frequenz-floss/frequenz-sdk-python/pull/117#pullrequestreview-1227363605

Reply:
If it is stuff internal to the power distributor, then I think it should live in `sdk.power` (or `actor.power_distribution` if this really is only intended to be used by the actor and nobody else) as internal modules `_battery_status.py` and `_battery_pool.py`. For me it is still OK to use the name if it makes sense, but it should be definitely very clear that it is some internal implementation detail of the power distributor then.

Reply:
No reason to reuse names and introduce ambiguities when we don't have to.

Reply:
BatteryPool is now used in PowerDistributor. But in the future it won't.

PowerDistirbutor will not track what batteries are working in future. It will just distribute power over given batteries.
User will use BatteryPool.set_power(...). And BatteryPool will use PowerDistributor.

Reply:
BatteryPool is used in PowerDistributor temporary. It will be extended to use the formulas and exposed to user.